### DISCUSSION TITLE
Engram without installing Keyman

### DISCUSSION THREAD

**Discussion by iklarman** [binarybottle/engram-layout#33] [Ideas]
Will there be a native MacOS & Windows install package at some point? (like how every other keyboard layout has)

**Comment by binarybottle** [comment]
I would love to have that, but don't have experience with this.  Could you help?

**Reply by iklarman** [reply]
Yea I can make one for MacOS.

**Reply by binarybottle** [reply]
That would be fantastic!

**Reply by AbbyRead** [reply]
Yeah, I'd be interested in this too.  Typing programs like GNU Typist and Tipp10 don't properly recognize the keys being pressed.  Klavaro works though, surprisingly.

**Reply by AbbyRead** [reply]
I made one for Windows using the Microsoft Keyboard Layout Creator.  All the relevant stuff is in this zip file, including the source, which is the .klc.  Should be able to open it and edit it if need be for any reason.  To install the layout, either click setup.exe or the .msi file specific to your computer's processor.  Setup.exe just seems to figure it out for you.  You can use the same msi to uninstall or repair.  
[Engram v2.0 Windows.zip](https://github.com/binarybottle/engram/files/8646337/Engram.v2.0.Windows.zip)
 
Hope this helps.  I'm gonna try making one for MacOS over vacation.  I'm new to this, but I really like the layout and don't ever want to have to switch back for any reason.  So when I tried playing TIS-100 and that game wasn't havin' it, I was like, "F*ck it, we'll do it live!"

**Reply by AbbyRead** [reply]
I claim no ownership rights.  Use however you like.  Thank you for your awesome work on this layout.

**Reply by AbbyRead** [reply]
Okay, some followup on further testing:
- I may have jumped the gun on thinking I needed to re-add QWERTY shortcuts, and the way I did it here caused some inconsistency with how certain programs handled the built-in functionality that I tried to override (needlessly, I think).
- I made a version without the Ctrl/Alt key assignment tomfoolery that I've labelled B for baseline.  So far no issues.
- Trying to use a program like DOSBox with any non-standard keyboard just defaults the keys back to QWERTY regardless.  I think this is due to whatever framework they are using (SDL, I think) opting for hardware key assignments over operating system native compatibility.  Probably simpler to do things that way, but it really throws a wrench in this stuff for me.  Not only did I have to reassign everything manually, but it actually wouldn't LET me split the characters from their original shift-held assignments.  And honestly, I don't know how well different shift/non-shift keybindings across the whole keyboard would work in tandem for games that require lots of keys held or unheld.  Trying to get into stenography with nkey rollover and chorded presses also makes me worried about that shit coming bact to haunt me.  So I came up with a Version C as in Compatible.  I honestly don't mind the way you laid it out in Engram 2.0; I'm just trying to make my life simpler while still sticking with the general layout.
- The three different ones all show up as the same name for me, despite my best efforts.  Windows, mang.  I dunno.  They install just fine though.

All three options (2b recommended for authenticity to creator's design, 2c if it's all too much to handle):
[Native Windows Keyboard Layout Options.zip](https://github.com/binarybottle/engram/files/8806807/Native.Windows.Keyboard.Layout.Options.zip)

C Layout:
![Engram2C](https://user-images.githubusercontent.com/77890127/171215922-993f3993-9c11-4c60-af0a-7822b3c0dc2b.jpg)
![Engram2CShft](https://user-images.githubusercontent.com/77890127/171215955-f1c1d680-0835-426b-ab1d-56357f37bb38.jpg)

**Reply by binarybottle** [reply]
Thank you for the Windows versions, @AbbyRead!  

Would you mind providing a brief README file?  For consistency, I would prefer to put the traditional Engram v2.0 you made in the install folder, and have its README point to this discussion with a description of how how the two variants differ.

**Reply by AbbyRead** [reply]
No problem!  I still want to make sure I'm on the right track with 2b, as I think the original version I shared was probably too hastily uploaded.  Just one of those, "Yay! I did something mildly helpful!" and it burning a hole in my pocket.  

I'll make sure it works on my end across things because I think we might not need the original version that I shared at all.  And if that's the case, even better.  Having a bunch of choices doesn't necessarily add value and could confuse people just looking to try _the_ layout. 

So if I can I want to get this right.  Also will be the first time I've made a readme, I think.  Sounds fun.

**Reply by binarybottle** [reply]
Thank you -- I appreciate it!

**Reply by AbbyRead** [reply]
[Native Windows Keyboard Layout Options.zip](https://github.com/binarybottle/engram/files/8858750/Native.Windows.Keyboard.Layout.Options.zip)
Okay, this should be the final version of the releases, readme included inside.  I recommend deleting the original version I uploaded and subsequently shit-talked about here (folder was "engram_2" I think).  I wrote the readme toward the end of the night today, so it's... interesting.

Hand-slappin' my way out of here for now, so do with the stuff as you please.  Thanks again for the awesome layout that I continue (today even) to recommend to any person who seems receptive.  👍🏻 ✌🏻

**Reply by binarybottle** [reply]
Love it!  I corrected some typos and added your name at the top of the README, but am otherwise keeping intact.  

"Hand-slappin' my way out of here" is one of my favorite lines ever!
